### PR TITLE
Consistent data plane code style 

### PR DESCRIPTION
**Pull request by pierDipi**
## Proposed Changes

- Add Checkstyle config
- Format (using https://github.com/google/google-java-format) and fix style errors

### CODE REVIEW

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java [hunk 1]
```diff
@@ -54,13 +55,12 @@ private ResourcesReconcilerImpl(
...
-      newResources
-        .stream()
-        .collect(Collectors.toMap(DataPlaneContract.Resource::getUid, Function.identity()))
-    );
+  public Future<Void> reconcile(final Collection<DataPlaneContract.Resource> newResources) {
+    final Map<String, DataPlaneContract.Resource> newResourcesMap =
+        new HashMap<>(
+            newResources.stream()
+                .collect(
+                    Collectors.toMap(DataPlaneContract.Resource::getUid, Function.identity())));
```
Comment (slinkydeveloper) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:63]: That style is so bad, with all the parenthesis at the end. I want lines with one close parens indented.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java [hunk 2]
```diff
@@ -75,67 +75,11 @@ private ResourcesReconcilerImpl(
...
-            futures.add(
-              this.ingressReconcilerListener.onDeleteIngress(oldResource, oldResource.getIngress())
-            );
-          } else if (newResource.hasIngress()) {
-            futures.add(
-              this.ingressReconcilerListener.onUpdateIngress(newResource, newResource.getIngress())
-            );
-          }
-        }
+        reconcileIngress(futures, oldResource, newResource, resourceAreEquals);
```
Comment (slinkydeveloper) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:80]: Why that change? The linter doesn't like big functions? I don't like that, big functions might have a performance reasoning behind it, i would prefer to remove that lint

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventbus/ContractMessageCodec.java [hunk 3]
```diff
@@ -22,26 +22,24 @@
 
 /**
  * This is a noop codec to send the contract on the event bus (works only in local environments)
- * <p>
- * https://github.com/eclipse-vertx/vert.x/issues/3375
+ *
+ * <p>https://github.com/eclipse-vertx/vert.x/issues/3375
```
Comment (slinkydeveloper) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventbus/ContractMessageCodec.java:26]: p should have its own line here

File: data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/ReceiverVerticle.java [hunk 4]
```diff
@@ -37,20 +37,22 @@
   private HttpServer server;
   private MessageConsumer<Object> messageConsumer;
   private final Function<Vertx, RequestMapper<String, CloudEvent>> requestHandlerFactory;
-  private final Function<Handler<HttpServerRequest>, Handler<HttpServerRequest>>[] handlerDecoratorFactories;
+  private final Function<Handler<HttpServerRequest>, Handler<HttpServerRequest>>[]
+      handlerDecoratorFactories;
```
Comment (slinkydeveloper) [inline data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/ReceiverVerticle.java:41]: The forced wrap here doesn't make any sense, why should i have the field name on a newline?

File: data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/ReceiverVerticle.java [hunk 5]
```diff
@@ -63,34 +65,29 @@ public ReceiverVerticle(
...
   @Override
-  public void stop(Promise<Void> stopPromise) {
-    CompositeFuture.all(
-      server.close().mapEmpty(),
-      messageConsumer.unregister()
-    )
-      .<Void>mapEmpty()
-      .onComplete(stopPromise);
+  public void stop(final Promise<Void> stopPromise) {
+    CompositeFuture.all(server.close().mapEmpty(), messageConsumer.unregister())
```
Comment (slinkydeveloper) [inline data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/ReceiverVerticle.java:89]: Why it didn't kept the `all` indentation?

File: data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java [hunk 6]
```diff
@@ -317,18 +335,18 @@ private static ConsumerDeployerVerticle setUpDispatcher(final Vertx vertx, final
...
-
-    final Function<Vertx, RequestMapper<String, CloudEvent>> handlerFactory = v -> new RequestMapper<>(
-      vertx,
-      producerConfigs(),
-      new CloudEventRequestToRecordMapper(v),
-      properties -> KafkaProducer.create(v, properties),
-      mock(Counter.class),
-      mock(Counter.class)
-    );
+  private static ReceiverVerticle setUpReceiver(final Vertx vertx, final VertxTestContext context)
```
Comment (slinkydeveloper) [inline data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java:338]: It wrapped the `throws`?! that's bad

**Review by slinkydeveloper — COMMENTED** [review]
Ok My biggest concern with this code style goes around 2 points:

* The forced wrapping of class, fields and method signatures. Can we remove it and figure out by ourselves when it makes sense or not to wrap?
* The forced closed parenthesis on the same line. This just makes the code more unreadable to me

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java (hunk 1 above)
Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:63]: google-java-style format code in this way, no way to change it by design.

Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:63]: There are no other maintained tools like that out there afaik.

File: data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java (hunk 6 above)
Comment (pierDipi) [reply data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java:338]: google-java-format is design to give a way to change style, bad or good, it's consistent, like in Go

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java (hunk 2 above)
Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:80]: Cyclomatic complexity was really high, should I remove it? It seems much more readable now.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventbus/ContractMessageCodec.java (hunk 3 above)
Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventbus/ContractMessageCodec.java:26]: google-java-style format code in this way, no way to change it by design.

**Comment by pierDipi** [comment]
> - The forced wrapping of class, fields, and method signatures. Can we remove it and figure out by ourselves when it makes sense or not to wrap?
> - The forced closed parenthesis on the same line. This just makes the code more unreadable to me

This is what we do and it hasn't worked since we have a codebase completely inconsistent.
I prefer to have a style that is consistent.
Also, I don't want to have discussions on a PR around what is a good or a bad style (too much opinionated, that leads to what we have today), I want to have a tool that does this for me, like `gofmt`.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java (hunk 2 above)
Comment (slinkydeveloper) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:80]: TBH I personally don't like cyclomatic complexity as a measurement of good code. In this case, I rather prefer to have the full algorithm in a single place, more than cutted in several pieces. This way I personally think it increases the brain overload to grasp the full algorithm, because I must go through all the different indirections. Deciding whether we should create a function or not is a human job, not a tooling job.

Also, as a side note, from the JIT point of view, this code might be optimized in a different way than the original one.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java (hunk 1 above)
Comment (slinkydeveloper) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:63]: Oh really? This is really a no go for me.... Can't you just disable the parenthesis formatting?

Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:63]: Nope, we can't.

**Comment by slinkydeveloper** [comment]
> Also, I'd like to avoid having discussions on PRs around what is a good or a bad style (too much opinionated, that leads to what we have today), I want to have a tool that does this for me, like gofmt.

Java is by design more verbose and less opinionated than Golang in terms of syntax and semantics, hence the hope to use a very opinionated tool like `gofmt` doesn't bring any good results. As you see these tool you tried made some very "opinionated choices" which doesn't make sense in practice: the forced wraps makes sense for the sake of consistency, but practically they make the code more unreadable. 

I'm ok with trying to make the codebase a bit more consistent, but this is not Golang, I think there is a certain degree of liberty that has to be left to the developer to make the code more reasonable to him. For me indenting the code this way:

```java
someOutput = list.stream()
  .map(someLambda).filter(Objects::nonNull)
  .collect(Collectors.toList())
```

Or this way:

```java
someOutput = list.stream()
  .map(someLambda)
  .filter(Objects::nonNull)
  .collect(Collectors.toList())
```

Makes me reason about the code in different ways. In the first case, my brain sees filter as part of the same logical step of map, it doesn't give to it much importance, while in the other case I see filter as a separate step, which deserves more attention. The tooling doesn't care about that and wraps anyway. Same exact discussion goes for wraps of fields (which is absolutely unreasonable IMO), signatures and, very important IMO, parens.

And, last but not least, the reviewers should be the people who ultimately care about the inconsistencies in code style. 

Long story short: I'm ok with adding a tool to format the codebase, but that tool should not decide the way I want to show the code to a reader.

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java (hunk 1 above)
Comment (matzew) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java:63]:
```
                    Collectors.toMap(DataPlaneContract.Resource::getUid, Function.identity())));
```
is ugly to read - no?

**Comment by pierDipi** [comment]
No way to make everybody happy, I tried clang-format, it has similar problems, closing.
